flocker/node/agents/cinder.py

```diff
@@ -98,22 +116,64 @@ def wait_for_volume(volume_manager, expected_volume,
             )
 
 
+def _instance_uuid():
+    """
+    See http://wiki.christophchamp.com/index.php/Xenstore
+    $ sudo xenstore-read name
+    instance-6ddfb6c0-d264-4e77-846a-aa67e4fe89df
+
+    # This is how we can get the instance_uuid of this node.
+    # But to satisfy the current IBlockDeviceAPI.list API we'll need
+    # to match the OpenStack instance_uuid to the hostname (or soon,
+    # the Flocker Node UUD). How will we do that?
+    # We don't really want this implementation to have to query the
+    # OpenStack servers list and even if it could get that list, how
+    # is it going to match Flocker node UUIDs to OpenStack
+    # instance_uuid.
+    # Perhaps BlockDeviceVolume.host should have the OpenStack
+    # instance_uuid and there should be an extra
+    # IBlockDeviceAPI.is_local_volume method (or something)
```

Comment:
I think we will add a new interface method, probably `IBlockDeviceAPI.what_is_this_nodes_cloudy_identifier`, and make `BlockDeviceVolume` include a `cloudy_identifier` to represent what it's attached to.

This is enough information to let `BlockDeviceDeployer` figure out which volumes are local.